src/main/java/backend/Logic.java

```diff
@@ -247,13 +209,204 @@ public Model getRepo(String repoId) {
+     * @param filterTexts Filter expressions to process for metadata requests.
+     * @return Repo IDs and the corresponding issues in the repo requiring a metadata update.
+     */
+    private Map<String, List<TurboIssue>> tallyMetadataUpdate(List<FilterExpression> filterTexts) {
+        HashMap<String, List<TurboIssue>> toUpdate = new HashMap<>();
+
+        List<TurboIssue> allModelIssues = models.getIssues();
+
+        filterTexts.stream().filter(Qualifier::hasUpdatedQualifier).forEach(filterText -> {
+            List<TurboIssue> issuesInPanel = allModelIssues.stream()
+                    .filter(issue -> Qualifier.process(models, filterText, issue))
+                    .collect(Collectors.toList());
+
+            issuesInPanel.forEach(issue -> {
+                List<TurboIssue> issuesInRepo = toUpdate.get(issue.getRepoId());
+                if (issuesInRepo != null) {
+                    // If eixsts, just add the issue
+                    issuesInRepo.add(issue);
+                } else {
+                    // If not exists, create first then add
+                    issuesInRepo = new ArrayList<>();
+                    issuesInRepo.add(issue);
+                    toUpdate.put(issue.getRepoId(), issuesInRepo);
+                }
+            });
+        });
+
+        return toUpdate;
+    }
+
+    /**
+     * Retrieves metadata for given issues from the repository source, and then processes them for non-self
+     * update timings.
+     *
+     * @param repoId The repository containing issues to retrieve metadata for.
+     * @param issues Issues sharing the same repository requiring a metadata update.
+     * @return True if metadata retrieval was a success, false otherwise.
+     */
+    public CompletableFuture<Boolean> getIssueMetadata(String repoId, List<TurboIssue> issues) {
+        String message = "Getting metadata for " + repoId + "...";
+        logger.info("Getting metadata for issues "
+                + issues.stream().map(TurboIssue::getId).map(Object::toString).collect(Collectors.joining(", ")));
+        UI.status.displayMessage(message);
+
+        String currentUser = prefs.getLastLoginUsername();
+
+        return repoIO.getIssueMetadata(repoId, issues).thenApply(this::processUpdates)
+                .thenApply(metadata -> {
+                    String updatedMessage = "Received metadata from " + repoId + "!";
+                    UI.status.displayMessage(updatedMessage);
+                    models.insertMetadata(repoId, metadata, currentUser);
+                    return true;
+                })
+                .exceptionally(withResult(false));
+    }
+
+    /**
+     * Filters and sorts issues within the model according to the given filter expressions.
+     *
+     * @param filterTexts Filter expressions to process.
+     * @return Filter expressions and their corresponding issues after filtering and sorting.
+     */
+    private Map<FilterExpression, ImmutablePair<List<TurboIssue>, Boolean>>
```

Comment:
Noticed that this boolean value is propagated quite a long way to ListPanel to determine if an issue should show non-self comments. Is this required? Could we check in ListPanel to see if the filter has an updated qualifier, then decide there?

If the boolean value is needed, probably should factor this out into a class. I'm starting to think that all the ImmutablePairs and ImmutableTriples were a mistake... :worried:

Can probably refactor further with groupingBy depending on above.